metrics/src/recorder.rs

```diff
@@ -168,9 +223,19 @@ pub unsafe fn set_recorder_racy(recorder: &'static dyn Recorder) -> Result<(), S
 /// and drop the installed recorder when clearing.  Thus, any existing recorder will stay leaked.
 ///
 /// This method is typically only useful for testing or benchmarking.
+///
+/// # Warning
+///
+/// This function is unsound and must not be called during any readers reading or writers writing.
+/// The caller can cause readers and writers to race if they are in reading/writing while
+/// this function is called.
 #[doc(hidden)]
 pub fn clear_recorder() {
-    STATE.store(UNINITIALIZED, Ordering::SeqCst);
+    // SAFETY: We cannot guarantee this here :/
+    // FIXME: Fix this unsoundness - either have a full `RwLock` around `recorder`, or make this unsafe
```

Comment:
I'd say we should just drop these internal comments and mark the overall function `unsafe`. The warning section covers the unsafety and lets users know what they have to do. We're never going to have a lock around the global recorder.